eng/Tools.props

```diff
@@ -11,5 +11,7 @@
     <PackageReference Include="Microsoft.DotNet.Build.Tasks.Packaging" Version="$(MicrosoftDotNetBuildTasksPackagingVersion)" />
     <PackageReference Include="Microsoft.NETCore.Platforms" Version="$(MicrosoftNETCorePlatformsVersion)" />
     <PackageReference Include="Microsoft.DotNet.VersionTools.Tasks" Version="$(MicrosoftDotNetVersionToolsTasksVersion)" />
+    <!-- Download the SDK in the initial restore step to work around race conditions when restoring in MSBuild -->
+    <PackageReference Include="Microsoft.DotNet.Helix.Sdk" Version="[$(MicrosoftDotNetHelixSdkVersion)]" />
```

Comment:
This should be a `PackageDownload` instead of a `PackageReference` since we just need to download it here.
```suggestion
    <PackageDownload Include="Microsoft.DotNet.Helix.Sdk" Version="[$(MicrosoftDotNetHelixSdkVersion)]" />
```

Reply:
In this spirit, should all these be package download?

Reply:
```msbuild
    <PackageDownload Include="Microsoft.DotNet.Build.Tasks.Packaging" Version="[$(MicrosoftDotNetBuildTasksPackagingVersion)]" />
    <PackageDownload Include="Microsoft.NETCore.Platforms" Version="[$(MicrosoftNETCorePlatformsVersion)]" />
    <PackageDownload Include="Microsoft.DotNet.VersionTools.Tasks" Version="[$(MicrosoftDotNetVersionToolsTasksVersion)]" />
    <!-- Download the SDK in the initial restore step to work around race conditions when restoring in MSBuild -->
    <PackageDownload Include="Microsoft.DotNet.Helix.Sdk" Version="[$(MicrosoftDotNetHelixSdkVersion)]" />
```

Sort of like this?

Reply:
The reason I suggest using `PackageDownload` is that is how CoreFX pulls down the IL SDK in their Tools.props.

https://github.com/dotnet/corefx/blob/4c3bd8a0533a2e35df371b91045a1a5f0ada7d54/eng/Tools.props#L18

Reply:
The rest of the packages that aren't SDKs, they still use `PackageReference`.

Reply:
I wonder why? They're not quite used as references. I don't feel like there's any difference in the usage case here. I don't feel strongly so I'll change it.